agent/openvas_agent.py

```diff
@@ -96,23 +98,35 @@ def _prepare_target(self, message: m.Message) -> str:
             target = self._prepare_url_target(message)
             return target
 
-    def _prepare_domain_target(self, message: m.Message) -> str:
+    def _prepare_domain_target(self, message: m.Message) -> Union[str, None]:
         """Prepares and checks if a domain asset has been processed before."""
         target = message.data.get('name')
         if not self.set_add(STORAGE_NAME, target):
             logger.info('target %s was processed before, exiting', target)
             return None
         else:
-            return target
-
-    def _prepare_url_target(self, message: m.Message) -> str:
+            schema = self._get_schema(message)
+            port = self.args.get('port')
+            if schema == 'https' and port not in [443, None]:
+                url = f'https://{target}:{port}'
+            elif schema == 'https':
+                url = f'https://{target}'
+            elif port == 80:
+                url = f'http://{target}'
+            elif port is None:
+                url = f'{schema}://{target}'
+            else:
+                url = f'{schema}://{target}:{port}'
+
+            return target if self._should_process_url(self._scope_urls_regex, url) else None
```

Comment:
should processs should not be in the prepare method as it is not expected as part of domain preparation. Just call it outside of the function to be clearer.